pkg/ruler/api.go

```diff
@@ -596,6 +639,14 @@ func (a *API) DeleteRuleGroup(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if a.ruler.IsNamespaceProtected(userID, namespace) {
+		if err = AllowProtectionOverride(req.Context(), namespace); err != nil {
+			level.Error(logger).Log("msg", "unable to create rule group under namespace", "err", err.Error())
```

Comment:
```suggestion
			level.Error(logger).Log("msg", "unable to delete rule group under namespace", "err", err.Error())
```

Reply:
Fixed in https://github.com/grafana/mimir/pull/8444/commits/22b03b83388c77f74628c1b246e38860f78e8d6d